src/core/network/webhook.ts

```diff
@@ -8,43 +8,39 @@ export default function (
   updateHandler: (update: Update, res: http.ServerResponse) => Promise<void>,
   errorHandler: (err: SyntaxError) => void
 ) {
-  return (
+  return async (
     req: http.IncomingMessage,
     res: http.ServerResponse,
-    next?: () => void
-  ): void => {
-    debug('Incoming request', req.method, req.url)
-    if (req.method !== 'POST' || req.url !== hookPath) {
-      if (typeof next === 'function') {
-        return next()
-      }
+    next = (): void => {
       res.statusCode = 403
       return res.end()
     }
+  ): Promise<void> => {
+    debug('Incoming request', req.method, req.url)
+    if (req.method !== 'POST' || req.url !== hookPath) {
+      return next()
+    }
     let body = ''
-    req.on('data', (chunk: string) => {
-      body += chunk.toString()
-    })
-    req.on('end', () => {
-      let update: Update
-      try {
-        update = JSON.parse(body)
-      } catch (error) {
-        res.writeHead(415)
+    for await (const chunk of req) {
+      body += String(chunk)
+    }
+    let update: Update
+    try {
+      update = JSON.parse(body)
+    } catch (error) {
+      res.writeHead(415)
+      res.end()
+      return errorHandler(error)
+    }
+    try {
+      await updateHandler(update, res)
+      if (!res.writableEnded) {
         res.end()
```

Comment:
Probably unnecessary, but too afraid to remove, not tested.

Reply:
It's actually covered by `webhookReply` tests I redid recently, and they still pass without it; removing.

Reply:
Tests can never show the absence of bugs, only their presence.

Can you explain why we don't have to close the connection? :thinking:

Reply:
If I see this correctly, then `end` is not called if the call to `getMe` fails. (This is not covered by tests.)

Reply:
Reintroduce the finally block you removed and this will be fixed.

Reply:
Ah no, the `catch` should do it, then I guess it's fine … this is confusing